### PR TITLE
Glob source

### LINKED ISSUE #200: Can't add wildcards into sass() call
I have used the new formatting as using `gulp.src` throws errors

```javascript
sass('src/scss/**/*.scss', { style: 'compressed' })
	.pipe( gulp.dest('build/css') );
```

but get the following error
```
[21:28:18] gulp-ruby-sass stderr: Errno::ENOENT: No such file or directory - src/scss/**/*.scss
  Use --trace for backtrace.
```

if i change to link to a single file eg. `src/scss/main.scss` it compiles fine, any ideas?

### PR DESCRIPTION
**Pull request by robwierzbowski**
Requested, and provided. Fixes #200.

GLOB

This PR adds single glob source support,  like `sass(source/**/*.scss)`. It filters out all `_partials`, which is probably desired behavior but we'll see once it's in the wild.

This comes on the heels of my test refactor/rewrite. It includes additional test refactoring and hardening which probably could be its own PR but I don't want to go crazy with rebases right now.

Code review and testing requested!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -54,44 +56,50 @@ function gulpRubySass (source, options) {
...
-		// sass's single file compilation doesn't create a destination directory, so
-		// we have to ourselves
-		mkdirp(intermediateDir);
-	}
+			if (path.extname(dest) !== '') {
+				dest = gutil.replaceExtension(dest, '.css');
+			}
 
-	// TODO: implement glob file source
+	    // sass options need unix style slashes, even if run on Windows
```
Comment (UltCombo) [inline index.js]: Perhaps we could remove the Windows slashes conversion. IIRC, I've added it to this plugin quite some time ago in order to workaround a bug in Sass, but it has already been fixed long ago.
/cc @chriseppstein

Comment (robwierzbowski) [inline index.js]: If we can test on a windows machine and get the version of Sass where this was fixed we should totally remove!

**Comment by robwierzbowski** [comment]
@UltCombo Just FYI, I'm refactoring tests to be more atomic (and just better), then adding tests to this branch.

File: index.js (hunk 1 above)
Comment (UltCombo) [inline index.js]: Looks like the issue we were working around has been fixed a year ago, see: https://github.com/sass/sass/issues/1328

Comment (UltCombo) [inline index.js]: It was fixed in Sass 3.3.14.

Comment (robwierzbowski) [inline index.js]: Awesome. We require Sass 3.4, so we can definitely remove. Adding a TODO on the issue.

Comment (robwierzbowski) [inline index.js]: Fixed!

Comment (UltCombo) [inline index.js]: Nice. :smile:

**Comment by huynhquancam** [comment]
@robwierzbowski Awesome :+1:

**Comment by robwierzbowski** [comment]
Mergin'!

**Pull request merged**